src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateArray = [];
+  const stateModif = { ...state };
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(stateModif, i.extraData);
+      stateArray.push({ ...stateModif });
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing a copy of the object to the array in each case. It's better to do this just once afterwards, outside of the `switch` block, after the current action is processed to avoid code repetition.